README.md

```diff
@@ -66,139 +69,222 @@ or
 
-   ```elisp
-   (use-package ts-fold-indicators
-      :load-path "/path/to/lib")
-   ```
+```elisp
+;; Example of ts-fold-range-alist's structure
+'((c-mode . c-folding-definitions) ;; *-folding-definitions is structured as shown below
+  (css-mode . css-folding-definitions)
+  (go-mode . go-folding-definitions)
+  (scala-mode . scala-folding-definitions)
+  ...)
+
+;; Examle of a folding definition alist
+(setq css-folding-definitions
+    (block . ts-fold-range-seq)
+    (comment . ts-fold-range-c-like-comment))
+```
 
-You can then enable this manually by doing the following
+So you can select whatever node that you want to fold on it.
 
-```
-M-x ts-fold-indicators-mode
-```
+To find what node you'll want to fold closed, refer to the (tree-sitter documentation)[https://emacs-tree-sitter.github.io/getting-started/#view-the-syntax-tree] about viewing nodes. `tree-sitter-debug` and `tree-sitter-query-builder` are both very useful for this.
 
-- To enable this automatically whenever `tree-sitter-mode` is enabled:
+For the folding functions, ts-fold provides some default
 
-   ```el
-   (add-hook 'tree-sitter-after-on-hook #ts-fold-indicators-mode)
-   ```
+- `ts-fold-range-seq` - Folds from the start of the node to the end of the node leaving a buffer of one character on each side. Usually used for code blocks that have bracketing delimiters.
 
-- To switch to left/right fringe: (Default is `left-fringe`)
+  ```c++
+  int main() { // <-- start of tree-sitter block node
+      printf("Hello, World\n");
+      return 0;
+  } // <-- end of tree-sitter block node
 
-   ```el
-   (setq ts-fold-indicators-fringe 'right-fringe)
-   ```
+  // |
+  // | '(block . ts-fold-range-seq)
+  // V
 
-- To lower/higher the fringe overlay's priority: (Default is `30`)
+  int main() {...} // Folded node
+  ```
 
-   ```el
-   (setq ts-fold-indicators-priority 30)
-   ```
+- `ts-fold-range-block-comment` - Folds multi-line comments that are of the form `/*...*/`. Should show a summary if the commentary plugin is turned on.
 
-- To apply different faces depending on some conditions: (Default is `nil`)
+  ```c++
+  /*
+   * The main function that gets run after program is compiled
+   * Doesn't take any parameters
+   */
+  int main() {
+      printf("Hello, World\n");
+      return 0;
+  }
 
-  For example, to coordinate [line-reminder](https://github.com/emacs-vs/line-reminder)
-with this plugin.
+  // |
+  // | '(comment . ts-fold-range-block-comment)
+  // V
 
-   ```elisp
-   (setq ts-fold-indicators-face-function
-      (lambda (pos &rest _)
-        (let ((ln (line-number-at-pos pos)))
-          (cond
-           ((memq ln line-reminder--change-lines) 'line-reminder-modified-sign-face)
-           ((memq ln line-reminder--saved-lines) 'line-reminder-saved-sign-face)
-           (t nil)))))
-   ```
+  / <S> * The main function that gets run after program is compiled *
```

Comment:
I think this should be `/* <S> The main function that gets run after program is compiled */`.

Reply:
My bad, my personal fold got reapplied when I was testing. However, when I just re-tested, I got 
`/* <S> * The main function that gets run after program is compiled *`. I'll update it to that for now. Can you confirm if that second asterisk is there for you too?

Reply:
I have tested in a `c++` file, the second asterisk isn't there... 😕 Just simply `/* <S> The main function that gets run after program is compiled */`.

Reply:
Hmm, strange but oh well 😆. Maybe something in my configuration is messing with it. I'll update the asterisk real quick.

Reply:
Yeah, this isn't blocking, and feel free to open up issues if that persist. It would be great if we can fix that! 😅

Reply:
Eh, I don't think it's worth diving into for the time being since there are other places for effort to be spent. And it really isn't blocking or troublesome 😛